File: objective-c/objc-1/h10.parent/Student.m
#import "Student.h"

@implementation Student

/********* INITIALIZERS ********/
- (Student *) init: (const char *) n gpa:(double) g {
    self = [super init];
    if (self) { self.name = n; self.gpa = g; }
    return self;
}

/********** PROPERTIES **********/
@synthesize gpa = _gpa;

@end

File: javascripts/js/h20.super/Student.js
load('Person.js');
/******* CLASS DEFINITION *******/
var Student = (function () {
  Student.prototype = new Person();

  /******* CONSTRUCTOR *******/
  function Student (name, gpa) {
    Person.call(this, name);
    this.gpa = gpa != null ? gpa : 0.0;
  }

  /******* PROPERTIES *******/
  Object.defineProperty(Student.prototype, "gpa", {
    get: function(){ return this._gpa },
    set: function(value) { this._gpa = value }
  });

  return Student;
})(Person);
